calc.py

```diff
@@ -0,0 +1,56 @@
+import math
+
+
+def add(a, b):
+    check_types(a, b)
+    return a + b
+
+
+def subtract(a, b):
+    check_types(a, b)
+    return a - b
+
+
+def multiply(a, b):
+    check_types(a, b)
+    return a * b
+
+
+def log(a):
+    check_type(a)
+    return math.log(a)
+
+
+def log10(a):
+     if a > 0:
+      check_type(a)
+      return math.log10(a)
+     else:
+      raise TypeError('input parameter must be positive number')
```

Comment:
When a is negative, TypeError wouldn't be the correct error to raise, since a negative number is still an int or float. ValueError would be the correct error to raise here.